### DISCUSSION TITLE
Model relation binding to a model list

### DISCUSSION THREAD

**Discussion by akauhovs** [livewire/livewire#3861] [1. Help]
What seems to be the problem:

Docs gives an example with model relation binding using one single model, that is simply binded in a mount() for example
and used in rules as user.posts.*.title
[](https://laravel-livewire.com/docs/2.x/properties#binding-models)

How to bind relations to a user list. so it would be as

component

```
public $bills;

protected $rules = [
        'bills.*.id' => 'nullable',
        'bills.*.name => 'nullable',
        'bills.payments.*.amount' => ['required','numeric'],
    ];

public function render()
    {
 $this->bills = PaymentBill::with('payment_description','payments')->get();
return view('livewire.admin.payment-list');
}
```

Blade

```
@foreach($bills as $index => $bill)

<div wire:key="bill-field-{{ $bill->id }}">
<input wire:model="bills.{{ $index}}.name" > // this is ok, I get the input
</div>

@foreach($bill->payments as $i => payment)

<div wire:key="payment-field-{{ $payment->id }}">
<input wire:model="bills.payments.{{ $i }}.amount"> // cant get input here
</div>

@endforeach

@endforeach
```
Mostly it is because PaymentBill model is not binded. How to bind model to each iteration of a model list ?

Thank you in advice

upd.
I can get data like $bills[0]->payments[0]->amount and so on.
but adding rule 
`'bills.*.payments.*.amount' ` doesn`t sovle the problem

**Comment by joshhanley** [comment]
@akauhovs yeah there are two pieces you need, in your rules, change it to `'bills.*.payments.*.amount'` as you suggested.

But you also need to update your `wire:model` binding in your view.

The payments amount input should be bound like this

```blade
<input wire:model="bills.{{ $index }}.payments.{{ $i }}.amount">
```

As having `$index` and `$i` in the same line can be confusing, I like to name them explicitly like `$billIndex` and `$paymentIndex`.

Hope this helps!

**Reply by bruno-fernandes-silsa** [reply]
I'm trying to use this logic, but I'm getting problems with the paginate method instead of the get method. 

Does anyone found out how to use this paginating results?

Thanks in advance

**Reply by joshhanley** [reply]
@bruno-fernandes-silsa you can't bind to paginated data. Can you explain in more detail what exactly you are trying to achieve so maybe we can come up with a solution?
